### ISSUE TITLE
Approach the ACM citation style

### ISSUE THREAD

**Issue by tomFellner** [se2p/se2thesis#7]
As mentioned in https://www.acm.org/publications/authors/reference-formatting the citation style should be of a numerical nature. The current template provides an alphabetical style:

https://github.com/se2p/se2thesis/blob/53d1f6b8f6800c20747f5607ce1cf6cfe2d9b0de/se2packages.dtx#L762-L772

Furthermore is the option `maxnames` set to a seemingly random value of 100. THis may be changed to a value of 1 to cling to the standard of the Chair of Software Engineering ;)

I suggest the following changes to the above code:

```tex
    \PassOptionsToPackage
      {
        backend=biber,
        backref=true,
        giveninits=true,
        hyperref=auto,
        maxnames=1,
        minalphanames=3,
        sorting=nyt,
        style=numeric,
      } { biblatex }
```

**Comment by stephanlukasczyk** [comment]
I know that ACM favours the numeric style. Is there a particular reason, why we should change it (except that it requires fewer space)?  Personally, I prefer the `alpha`-like style, because it gives you an impression of what the referenced publication could be; `[LKF20]` is probably a more memorable reference that `[42]`.

Don't get me wrong, I am open for discussion, just want to know the ideas behind the proposal.

Regarding the `maxnames` value: yes, that is arbitrary, but usually large enough to prevent the (in my opinion ugly and also quite excluding) truncation. My problem with the truncation is that it is also arbitrary, where and when it should happen.  Should the references section contain something like `Maier et al. 2023...`, or `Maier, Müller, Lehmann. 2023...`, or `Maier, Müller, et al. 2023...`?  IIRC, ACM does not truncate the author names at all, does it?

**Comment by tomFellner** [comment]
As this package description states, this LaTeX package provides a good setup for writing theses for the [Chair of Software Engineering II](https://www.fim.uni-passau.de/en/chair-for-software-engineering-ii/) at the [University of Passau](https://www.uni-passau.de/). As my advisor suggests, I should use the `numeric` citation style.

At first, I thought this may be easily achievable through `\PassOptionsToPackage{style=numeric}{biblatex}`, but this does not seem possible: https://tex.stackexchange.com/a/267324

This lead me to the only possible solution - overwrite the `biblatex` package myself. An update on your side would not be applied to my paper as my package declaration wins.

The same argument applies to the `maxnames`.

Referring to the suggested citation style `numeric`: I can only guess which motivation lead the ACM to the citation style - fewer space may be one of them. I prefer it when reading papers, as at first I'm more interested in the content of the paragraphs and so I can ignore the `[\d+]` statements. If there are alphabetic characters involved, it gets harder to ignore. When searching for in depth information I might follow the hyperrref to the bibliography indenpendent of the displayed characters.

Referring to the arbitrary `maxnames`: you're right - there is no suggested way defined by ACM in the above linked page. ACM provides a document class `acmart` https://github.com/borisveytsman/acmart which gives insight into their citation style: https://github.com/borisveytsman/acmart/blob/primary/acmnumeric.bbx. They set `maxcitenames=2`, which may be applicable to the `se2packages`.

In summary my suggestion derives from the requirements applied to my paper. I suggest the Chair of Software Engineering II to find common ground to reduce noise for the students ;)

**Comment by stephanlukasczyk** [comment]
OK, I see, you did your research homework well :wink: 

> I suggest the Chair of Software Engineering II to find common ground to reduce noise for the students ;)

We'll have some discussions and I'll come back to this issue afterwards.

**Comment by stephanlukasczyk** [comment]
Random thought, @tomFellner , since I'm playing around with this currently: why not using the ACM bibliography style directly?  You could set `biblatex=false` when loading `se2packages` and load `biblatex` manually:
```latex
\usepackage[
  backend=biber,
  backref=true,
  giveninits=true,
  hyperref=auto,
  sorting=nyt,
  datamodel=acmdatamodel,
  style=acmnumeric,
]{biblatex}
\DeclareFieldFormat{titlecase}{#1}
\newcommand{\bibliofont}{\normalsize}
```
Please note that the last line (definition of the `\bibliofont` macro) is necessary because it is not provided by standard LaTeX classes but only by the ACM document class. The second last line assures that capitalisation of titles is preserved.

I am now thinking to change the BibLaTeX loading code in `se2packages` to the aforementioned one and resolve all the issues with this :wink:

What do you think?

**Comment by tomFellner** [comment]
Hey @stephanlukasczyk, currently I'm not directly writing on my thesis - implementation comes first. I hope in a view days I can test your suggestion.
Your approach to use ACM bibliography style directly seems promissing to me, as this can reduce discussions.

I will stay in touch ;)
